File: Server/src/main/java/com/springboot/garageApp/service/GarageService.java
package com.springboot.garageApp.service;

import com.springboot.garageApp.model.Car;
import com.springboot.garageApp.repository.GarageRepository;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class GarageService {

    private GarageRepository garageRepository;

    public GarageService( GarageRepository garageRepository){
        this.garageRepository = garageRepository;
    }

    public List<Car> getCars() {
        return garageRepository.findAll();
    }

    public Car getCar(long id) {
        return garageRepository.findById(id);
    }

    public void deleteCar(long id) {
        garageRepository.deleteById(id);
    }

    public void addCar(Car car) {
        garageRepository.save(car);
    }

    public void updateCar(Car car, long id) {
        Car carInDB = garageRepository.findById(id);
        if(carInDB != null){
            if(car.getBrand() != null)
                carInDB.setBrand(car.getBrand());
            if(car.getColor() != null)
                carInDB.setColor(car.getColor());
            if(car.getYear() != 0)
                carInDB.setYear(car.getYear());
            if(car.getModel() != null)
                carInDB.setModel(car.getModel());
            garageRepository.save(carInDB);
        }
    }
}

File: Server/src/main/java/com/springboot/garageApp/service/GarageService.kt
package com.springboot.garageApp.service

import com.springboot.garageApp.model.Car
import com.springboot.garageApp.repository.GarageRepository
import org.springframework.stereotype.Service
import java.util.function.Consumer

@Service
class GarageService(private val garageRepository: GarageRepository) {
    val cars: List<Car>
        get() {
            val carList: MutableList<Car> = ArrayList()
            garageRepository.findAll().forEach(
                Consumer { car: Car? ->
                    if (car != null) {
                        carList.add(car)
                    }
                }
            )
            return carList
        }

    fun getCar(id: Long): Car? {
        return garageRepository.findById(id).orElse(Car())
    }

    fun deleteCar(id: Long) {
        garageRepository.deleteById(id)
    }

    fun addCar(car: Car) {
        garageRepository.save(car)
    }

    fun updateCar(car: Car, id: Long) {
        val carInDB = garageRepository.findById(id).get()
        if (carInDB != null) {
            if(car.brand != null) carInDB.brand = car.brand
            if(car.color != null) carInDB.color = car.color
            if(car.year != 0) carInDB.year = car.year
            if(car.model != null) carInDB.model = car.model
            garageRepository.save(carInDB)
        }
    }
}
